### PR TITLE
Semantic versioning for v7

### PR DESCRIPTION
**Pull request by faddat**
Works on: #788 

## Description

Semantic versioning for osmosis v7

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,9 +1,9 @@
-module github.com/osmosis-labs/osmosis
+module github.com/osmosis-labs/osmosis/v7
 
 go 1.17
 
 require (
-	github.com/cosmos/cosmos-sdk v0.44.5
+	github.com/cosmos/cosmos-sdk v0.45.0
```
Comment (mattverse) [inline go.mod:6]: We should probably remain on v0.44.5 for now!

**Review by mattverse — CHANGES_REQUESTED** [review]
Should be good to merge once go.mod sdk version is reverted to original state!

**Comment by ValarDragon** [comment]
I am generally lost as to what is standard for golang projects. Is it to make main with breaking changes with one version higher, or is it with the same last major release.

Also how do golang projects do backports with these version import differences :/

**Comment by mattverse** [comment]
From my knowledge its make breaking changes with one version higher.

**Comment by mattverse** [comment]
> Also how do golang projects do backports with these version import differences :/
@ValarDragon wdym? Can you give an example ?

**Comment by faddat** [comment]
You know, there's an issue in this, too.  I'm going to jump off of main a second time, to avoid it.

**Comment by faddat** [comment]
@ValarDragon if you still want to use mergify, we need to add it to this repo.  I am going to close this one because I think it might add commits of Sunny's not meant for `main`. 

Also seems like some of the wasm tests are trying to import /x/ibc

I think I got halfway through an upgrade of the tests w/ Juno and it should be the same kind of thing so I'll try and fix them.

**Comment by faddat** [comment]
@mattverse this is complete and I've merged the latest main branch now, too.

**Review by mattverse — APPROVED** [review]
LGTM! Thanks for this @faddat 

In the future, we don't want main to have sem-ver of the next major release, but sem-ver for the latest release, since that would make sense for all the backporting that would be happening. Since we have a major upgrade coming up soon, this PR, in particular, makes sense + super helpful!

File: x/lockup/keeper/legacy.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
+package keeper
```
Comment (ValarDragon) [inline x/lockup/keeper/legacy.go:1]: Lets re-delete this file

**Review by ValarDragon — APPROVED** [review]
Thanks for the in-depth review @mattverse / checking what other projects do as the standard here.

Did a quick skim, and only saw the v7 change + the re-added legacy.go file.

Lets also file an issue to add the semver situation to the contributing.md

**Comment by ValarDragon** [comment]
Going to go ahead and merge before this is broken by the next PR

**Pull request merged**